### PR TITLE
Stack effect

### PR DESCRIPTION
**Pull request by rabbitonweb**
Allows building/testing/executing Haskell stack based projects

### CODE REVIEW

File: src/Frost/Effects/Stack.hs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+runStack = interpret $ \case
+  Clean                       -> stack "clean"
+  Build                       -> stack "build"
+  Exec what                   -> stack $ "exec " ++ what
+  Test                        -> stack "test"
+  TestMatch specName testName -> stack $ "test --match " ++ (show $ matchStr specName testName)
+  where
+    stack :: Member Sys r => String -> Sem r String
+    stack arg = fmap snd (cmd $ "stack --no-terminal " ++ arg)
+    matchStr specName testName = "/Stack/" ++ specName ++ "/" ++ testName ++ "/"
```
Comment (lukasz-golebiewski) [inline src/Frost/Effects/Stack.hs]: Why the `/Stack` prefix?

File: src/Frost/Effects/Stack.hs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+makeSem ''Stack
+
+runStack :: Member Sys r => Sem (Stack ': r) a -> Sem r a
+runStack = interpret $ \case
+  Clean                       -> stack "clean"
+  Build                       -> stack "build"
+  Exec what                   -> stack $ "exec " ++ what
+  Test                        -> stack "test"
+  TestMatch specName testName -> stack $ "test --match " ++ (show $ matchStr specName testName)
```
[the remark this answers is not in the document]
Comment (rabbitonweb) [reply src/Frost/Effects/Stack.hs]: `runSysIO` needs integration tests, which handles error cases

File: src/Frost/Effects/Stack.hs (hunk 1 above)
Comment (rabbitonweb) [reply src/Frost/Effects/Stack.hs]: stack used to add `/Stack`, now it does not 🤷‍♂

Comment (rabbitonweb) [reply src/Frost/Effects/Stack.hs]: so it means that for now we only support latest stack

**Comment by rabbitonweb** [comment]
@lukasz-golebiewski this is ready for a second look

**Pull request merged**